File: lint-arwaky/archive/contract/arch_coordinator_aggregate.py
from abc import ABC, abstractmethod
from ..taxonomy import FilePath, ComplianceStatus, LintResultList

class ArchCoordinatorAggregate(ABC):
    """AGGREGATE: Orchestrates architectural compliance across the codebase."""
    @abstractmethod
    async def check_compliance(self, path: FilePath) -> ComplianceStatus:
        """Button: Perform full compliance check for a project path."""
        ...

    @abstractmethod
    async def scan(self, path: FilePath) -> LintResultList:
        """Bridge: Perform architectural scan for the linter pipeline."""
        ...

    @abstractmethod
    async def apply_fix(self, path: FilePath) -> ComplianceStatus:
        """Bridge: Apply architectural fixes."""
        ...

File: lint-arwaky/src-rust/contract/arch_coordinator_aggregate.rs
use async_trait::async_trait;
use super::*;

#[async_trait]
pub trait ArchCoordinatorAggregate: Send + Sync {
    async fn check_compliance(&self, path: &FilePath) -> ComplianceStatus;
    async fn scan(&self, path: &FilePath) -> LintResultList;
    async fn apply_fix(&self, path: &FilePath) -> ComplianceStatus;
}
